src/main/webapp/app/shared/utils/LodashUtils.ts

```diff
@@ -0,0 +1,273 @@
+    ([pass, fail], elem) =>
+      predicate(elem) ? [[...pass, elem], fail] : [pass, [...fail, elem]],
+    [[], []]
+  );
+}
+
+/**
+ * Sorts an array of objects by the specified key.
+ * @param array - The array to sort.
+ * @param key - The key to sort by.
+ * @returns The sorted array.
+ */
+export function sortByKey<T extends object & NonArrayObject, K extends keyof T>(
+  array: T[],
+  key: K
+) {
+  return array.sort((a: T, b: T) =>
+    a[key] > b[key] ? 1 : b[key] > a[key] ? -1 : 0
+  );
+}
+
+/**
+ * Sorts an array of objects by the result of the comparator function.
+ * @param array - The array to sort.
+ * @param comp - The function to determine the sort order.
+ * @returns The sorted array.
+ */
+export function sortBy<T extends object & NonArrayObject, K extends keyof T>(
+  array: T[],
+  comp: (datum: T) => number | string
+) {
+  return array.sort((a: T, b: T) => {
+    const ra = comp(a);
+    const rb = comp(b);
+    if (typeof ra === 'string' || typeof rb === 'string') {
+      return `${ra}`.localeCompare(`${rb}`);
+    } else {
+      return ra - rb;
+    }
+  });
+}
+
+/**
+ * Converts the first character of `string` to upper case and the remaining to lower case.
+ * @param string - The string to convert.
+ * @returns The converted string.
+ */
+export function upperFirst(string: string) {
+  return string
+    ? string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
+    : '';
+}
+
+/**
+ * Creates a duplicate-free version of an array.
+ * @param array - The array to inspect.
+ * @returns The new duplicate-free array.
+ */
+export function uniq<T>(array: T[]) {
+  return [...new Set(array)];
+}
+
+/**
+ * This method is like `uniq` except that it accepts `iteratee` which is invoked for each element in `array` to generate the criterion by which uniqueness is computed.
+ * @param array - The array to inspect.
+ * @param iteratee - The iteratee invoked per element.
+ * @returns The new duplicate-free array.
+ */
+export function uniqBy<T extends object & NonArrayObject>(
+  array: T[],
+  iteratee: ((item: T) => any) | keyof T
+) {
+  const seen = new Set();
+  const result = [];
+
+  for (const item of array) {
+    const computed =
+      typeof iteratee === 'function' ? iteratee(item) : item[iteratee];
+    if (!seen.has(computed)) {
+      seen.add(computed);
+      result.push(item);
+    }
+  }
+
+  return result;
+}
+
+/**
+ * Creates an array of `array` values not included in the other given arrays. The order of result values is determined by the order they occur in the arrays.
+ * @param arrays - The arrays to inspect.
+ * @returns The new array of filtered values.
+ */
+export function xor<K extends number | string>(...arrays: K[][]) {
+  // Manually flatten the arrays into a single array
+  const flattened = flatten(arrays);
+
+  // Create a frequency map to count occurrences of each element
+  const frequencyMap = flattened.reduce((acc, val: K) => {
+    acc[`${val}`] = (acc[`${val}`] || 0) + 1;
```

Comment:
If we have two arrays:
`[ '100', 2, 3] `
`[ 100, 2, 3] `
Using string formatting for the key will cause an issue here because `${100}` and `${'100'}` will both return `'100'`. I would expect xor will return `['100', 100]`?

Reply:
Methods in this util do not support mix types.